nhrpd/nhrp_peer.c

```diff
@@ -565,20 +565,18 @@ static void nhrp_handle_resolution_req(struct nhrp_packet_parser *pp)
 	while ((ext = nhrp_ext_pull(&pp->extensions, &payload)) != NULL) {
 		switch (htons(ext->type) & ~NHRP_EXTENSION_FLAG_COMPULSORY) {
 		case NHRP_EXTENSION_NAT_ADDRESS:
-
-			if (sockunion_family(&nifp->nat_nbma) == AF_UNSPEC)
-				break;
-
 			ext = nhrp_ext_push(zb, hdr,
```

Comment:
is it really mandatory ?
from https://www.cisco.com/c/en/us/td/docs/ios-xml/ios/sec_conn_dmvpn/configuration/15-mt/sec-conn-dmvpn-15-mt-book/sec-conn-dmvpn-dt-spokes-b-nat.html , it is written:
```
When a spoke is behind a NAT device, it includes a NAT NHRP extension when it sends NHRP resolution requests. ```

Reply:
Our reference is also the same doc you mentioned as well as the real cisco routers we've got. We've analyzed the Wireshark captures of packets coming from cisco during FRR-Cisco interop

What Cisco does is - if Cisco spoke is not behind NAT then still it includes an "empty" NAT extension header.

Reply:
Cisco includes a flag if a peer supports NAT extension in 'show ip nhrp' output. @amollad is the functionality of including empty NAT header is used to provide this information?

Reply:
the response from @amollad is ok for me, as the implementation relies also on the cisco wireshark captures.